backend/api/Controllers/MissionSchedulingController.cs

```diff
@@ -179,6 +179,23 @@ ISourceService sourceService
                 missionRun.CalculateEstimatedDuration();
             }
 
+            IList<MissionStatus> missionStatuses = [MissionStatus.Ongoing, MissionStatus.Pending, MissionStatus.Paused];
+            var returnToHomeMission = await missionRunService.ReadReturnToHomeMissionRuns(robot.Id, missionStatuses);
+            if (returnToHomeMission != null)
+            {
+                if (!await localizationService.RobotIsOnSameDeckAsMission(robot.Id, missionRun.Area.Id))
+                {
+                    return BadRequest($"The robot {robot.Name} is localized on a different deck so the mission was not scheduled.");
```

Comment:
Should we use 409 Conflict instead of BadRequest? It can be argued that the request is valid, since it can work exactly the same with a different state